src/v/utils/moving_average.h

```diff
@@ -50,6 +51,35 @@ class moving_average {
     size_t _valid_samples{0};
 };
 
+template<typename T, size_t Samples>
+class moving_max {
+public:
+    explicit moving_max(T initial_value)
+      : _running_max(initial_value) {
+        std::fill(_samples.begin(), _samples.end(), initial_value);
+    }
+
+    void update(T v) {
+        // Value at '_ids' will be phased out. If it's equal to current
+        // max we need to recalculate it.
+        bool need_recalc = _running_max == _samples[_idx];
+        _samples[_idx] = v;
+        _idx = (_idx + 1) % Samples;
```

Comment:
not a nit or suggestion, but I read recently that something like

```
auto x = _idx + 1;
_idx = x >= Samples ? x % Samples : x;
```

(maybe with `unlikely` added to the predicate condition) is faster for wrapping counters. But maybe the compiler handles this anyway and this would be a micro optimization.

Reply:
removed that code since history is no longer used